geode-benchmarks/src/main/java/org/apache/geode/benchmark/topology/ClientServerTopologyWithSniProxy.java

```diff
@@ -58,23 +64,29 @@ public static SniProxyImplementation valueOfIgnoreCase(String name) {
     }
   }
 
-  public static void configure(final TestConfig config,
-      final SniProxyImplementation sniProxyImplementation) {
+  public static void configure(final TestConfig config) {
     role(config, LOCATOR, NUM_LOCATORS);
     role(config, SERVER, NUM_SERVERS);
     role(config, CLIENT, NUM_CLIENTS);
     role(config, PROXY, NUM_PROXIES);
 
+    configureBefore(config);
+
+    before(config, new StartClientWithSniProxy(LOCATOR_PORT, SNI_PROXY_PORT, PROXY), CLIENT);
```

Comment:
Notice the subtlety of the router calling into here. It needs to start a client with a different proxy address. That last arg that is `PROXY` here is `ROUTER` in the router topology.

I could pull the StopClient out of the configuratAfter. Thoughts?

Reply:
I see that you're doing this because the subclass (`ClientServerToplogyWithSniProxy`) also wants to call `before()` but these two classes (this base and that subclass) need different client tasks ok ✓